### DISCUSSION TITLE
Chapter number as part of the image and table labeling in format html

### DISCUSSION THREAD

**Discussion by MarcoPortmann** [quarto-dev/quarto-cli#10011] [Q&A]
### Description

I have a report that I would like to provide in pdf and html formats. For the pdf, I use documentclass: scrreport. The figures and tables are numbered by chapter, e.g. "Figure 1.1", "Figure 2.1". In the html document, the same tables are numbered as "Figure 1" and "Figure 2". I know that I can achieve the desired numbering by switching to a quarto book project. However, this also seems to have disadvantages. For example, I would like to have the entire text in one file, be able to share all resources between chapters, and have only one TOC to navigate through the document.

Is there a way to include the chapter number in the format html in the labeling of the tables and figures?

**Comment by mcanouil** [comment]
There is currently no options for that, but you could use custom cross-reference to achieve your goal, see <https://quarto.org/docs/authoring/cross-references-custom.html>.

For reference:

- https://github.com/quarto-dev/quarto-cli/issues/1818
- https://github.com/quarto-dev/quarto-cli/issues/6502

**Reply by MarcoPortmann** [reply]
After some research, it seems to me that this could be possible with a Lua filter. Lua filters are new territory for me, but if I manage to write a filter, I'll be happy to post it.

**Reply by mcanouil** [reply]
Yes, a LUA filter that occurs after Quarto processing might work
Note that you will want your filter to work only for HTML format (either by adding a check in the filter, or by setting the filter under `format.html` in your YAML.

**Comment by mingsu** [comment]
This is what I expect, I manage my reports in a quarto web project, and write reports in different subfolders. 

Is it possible to include a quarto "book" type inside a quarto "web" type project?

**Reply by mcanouil** [reply]
@mingsu Your question seems quite off topic.
And your question has already a discussion:

- https://github.com/quarto-dev/quarto-cli/discussions/3505
